plugins/updater/src/updater.rs

```diff
@@ -1363,6 +1371,21 @@ impl PathExt for PathBuf {
     }
 }
 
+#[cfg(windows)]
+fn escape_nsis_current_exe_arg(arg: &&OsStr) -> String {
+    let mut arg = arg.to_string_lossy().to_string();
+
+    /* if arg.contains('"') {
+        arg = arg.replace('"', r#""""#);
+    } */
+
+    if arg.contains(' ') || arg.contains('\t') || arg.contains('/') {
+        arg = format!("\"{arg}\"");
+    }
+
+    arg
```

Comment:
```suggestion
    let arg = arg.to_string_lossy().to_string();
    if arg.contains(' ') || arg.contains('\t') || arg.contains('/') {
        format!("\"{arg}\"")
    } else {
        arg
    }
```

Reply:
That's a reminder. Couldn't test " in args yet (well I tried but rusts args parsing is weird (It's probably me))

Reply:
I see, after some testing, I think only the one from the std really worked, basically, you need to escape `"` as `\"` (it also handles the `\"` case to `\\\"` I believe)

```suggestion
    let mut cmd: Vec<char> = Vec::new();

    let quote = arg.chars().any(|c| c == ' ' || c == '\t' || c == '/') || arg.is_empty();
    let escape = true;
    if quote {
        cmd.push('"');
    }

    let mut backslashes: usize = 0;
    for x in arg.chars() {
        if escape {
            if x == '\\' {
                backslashes += 1;
            } else {
                if x == '"' {
                    // Add n+1 backslashes to total 2n+1 before internal '"'.
                    cmd.extend((0..=backslashes).map(|_| '\\'));
                }
                backslashes = 0;
            }
        }
        cmd.push(x);
    }

    if quote {
        // Add n backslashes to total 2n before ending '"'.
        cmd.extend((0..backslashes).map(|_| '\\'));
        cmd.push('"');
    }
    cmd.into_iter().collect()
```

Reply:
Have a good rest haha

My problem in testing was that I couldn't even get the " into the initial app instance's env::args() call lol
I think I messed up my Powershell profile _again_

Anyway, will get back to this tomorrow

Reply:
try with `app.exe '"some args"'`, `'` escapes things inside in powershell

Reply:
Yeah that's what I tried first though can't remember if I typed something inbetween.

Reply:
okay either my powershell instance is broken beyond repair or that std args parser is missing a few edge cases (i may be testing stuff that aren't ever used idk). will just use this and call it a day.